src/structurizr/model/software_system_instance.py

```diff
@@ -67,15 +67,11 @@ def hydrate(
         model: "Model",
         parent: "DeploymentNode",
     ) -> "SoftwareSystemInstance":
-        """Hydrate a new SoftwareSystemInstance instance from its IO.
-
-        This will also automatically register with the model.
-        """
+        """Hydrate a new SoftwareSystemInstance instance from its IO."""
         system = model.get_element(system_instance_io.software_system_id)
```

Comment:
Could the system be provided as an argument to `hydrate` instead of the model?

Reply:
I'm in two minds about this, as it'd mean that the `DeploymentNode` would be pulling stuff out of the `SoftwareSystemInstanceIO` (and similarly for `ContainerInstance`) which feels slightly odd, but then again it is also cleaner from a dependency-injection viewpoint as `SoftwareSystemInstance` wouldn't need to call out to anything.

I'll give it a go and we can see how it looks.

Reply:
OK, that's committed now so see if that looks cleaner (I think it does).